controllers/Items.php

```diff
@@ -170,4 +170,48 @@ public function stats() {
 
         $this->view->jsonSuccess($stats);
     }
+
+
+    /**
+     * returns updated database info (stats, item statuses)
+     * json
+     *
+     * @return void
+     */
+    public function sync() {
+        $this->needsLoggedInOrPublicMode();
+
+        if( !array_key_exists('since', $_GET) )
+            $this->view->jsonError(array('sync' => 'missing since argument'));
+
+        $since = new \DateTime($_GET['since']);
+
+        $itemsDao = new \daos\Items();
+        $last_update = new \DateTime($itemsDao->lastUpdate());
+
+        $sync = array(
+            'last_update' => $last_update->format(\DateTime::ISO8601),
+        );
+
+        if( $last_update > $since ) {
+            $sync['stats'] = $itemsDao->stats();
+
+            if( array_key_exists('tags', $_GET) && $_GET['tags'] == 'true' ) {
```

Comment:
Now this is almost like GraphQL. Would not it be better to create a GraphQL endpoint? It could also save a lot requests during initial connection.